did-sdk/src/key/key.ts

```diff
@@ -0,0 +1,120 @@
+import * as key from '@transmute/did-key.js'
+import { DIDDocument } from 'did-resolver'
+import { ReadOnlyDidDocAPI, WritableDidDocAPI } from '../did-documents.js'
+
+export type generateFromSeedOptions = { secureRandom: () => Buffer }
+export type generateFromRandomOptions = {
+  kty: 'EC' | 'OKP'
+  crvOrSize: 'P-256' | 'P-384' | 'P-521'
+}
+export type KeyType =
+  | 'ed25519'
+  | 'x25519'
+  | 'secp256k1'
+  | 'bls12381'
+  | 'secp256r1'
+  | 'secp384r1'
+  | 'secp521r1'
+type Unpacked<T> = T extends (infer U)[] ? U : T
+type GeneratedDid = Awaited<ReturnType<typeof key.generate>>
+export type DidKey = Unpacked<GeneratedDid['keys']>
+export type DidKeySerializedState = {
+  id: string
+  keys: DidKey[]
+}
+
+export async function resolve(didUri: string): Promise<ReadOnlyDidDocAPI> {
+  const doc = await key.resolve(didUri, {
+    accept: 'application/did+ld+json',
+  })
+  return new ReadOnlyDidDocAPI(doc.didDocument)
+}
+
+export async function create(
+  type: KeyType,
+  generateOptions?: generateFromSeedOptions | generateFromRandomOptions,
+): Promise<KeyDidDocAPI> {
+  const did = new KeyDidDocAPI()
+  await did.create(type, generateOptions)
+  return did
+}
+
+export async function inst(
+  state: DidKeySerializedState,
+): Promise<KeyDidDocAPI> {
+  const did = new KeyDidDocAPI()
+  await did.hydrate(state)
+  return did
+}
+
+export class KeyDidDocAPI extends WritableDidDocAPI {
+  id = ''
+  keys: DidKey[] = []
```

Comment:
is there ever a world in which a `did:key` has more than one key? seems to go against the nature of it 🤔 

or is this just for consistency's sake?

Reply:
This is because the module we're using generates both the ed25519 and the derived x25519.

Reply:
ahhh got it